pkg/models/plan.go

```diff
@@ -1,32 +1,46 @@
 package models
 
 import (
-	"github.com/bacalhau-project/bacalhau/pkg/jobstore"
 	"github.com/bacalhau-project/bacalhau/pkg/model"
 )
 
+type PlanExecutionDesiredUpdate struct {
+	Execution    *model.ExecutionState
+	DesiredState model.ExecutionDesiredState
+	Comment      string
+}
+
 // Plan holds actions as a result of processing an evaluation by the scheduler.
 type Plan struct {
```

Comment:
While this sounds nice, I am afraid it is adding a lot of complexity that I am not sure I agree with the all the benefits. `Plan` is the object and model that contains all the mutations and updates that planners must execute. There shouldn't be multiple implementation for a`Plan`, which means we shouldn't define it as in interface. 

I understand the desire of marking the `Plan` immutable to avoid accidental modifications, but this doesn't justify introducing interfaces. If the concern is with `AppendX` and `MarkJobX` methods, then I can move them outside of `Plan` to a set of helper methods

Reply:
Would you be open to the suggested alternative then? Make the fields private? This will at least make it immutable outside the package.